drivers/hmis/app/models/hmis/hoh_change_handler.rb

```diff
@@ -0,0 +1,113 @@
+module Hmis
+  #
+  # The HohChangeHandler class is responsible for handling changes to the Head of Household (HoH) in a household enrollment.
+  # It provides methods to perform the change, apply the changes to the database, and validate the changes.
+  #
+  # This logic is split out from the UpdateRelationshipToHoH mutation in case we need to use it in cleanup routines,
+  # for example bulk assigning HoH to the oldest adult in a household.
+  #
+  # Parameters:
+  # - new_hoh_enrollment: The Enrollment that is becoming the Head of Household
+  # - hud_user_id: The "UserID" of the HUD user making the change.
+  class HohChangeHandler
```

Comment:
Is this the right location/naming for this? We have some other non-model classes in `drivers/hmis/app/models` too, like the enrollment reminder generator.